### PR TITLE
Added Cypress Test for MM-11390 Guest Invitation Flow: Verify Add New/Existing Guest Users

### PR DESCRIPTION
**Pull request by srkgupta**
#### Summary
Validating the Messages in Guest Invitation Flow when a user adds a new or existing user
1. Click on Invite People and then Invite Guest Users
2. Search and add an existing member by username, who is part of the team and click Invite Guests and verify the error message 'This person is already a member.' is displayed
3. Search and add an existing member by email, who is not part of the team and click Invite Guests and verify the error message 'This person is already a member.' is displayed
4. Demote a regular member of a team to a guest user and then search and add an existing guest by first name, who is part of the team but not channel and click Invite Guests and verify the success message ‘This guest has been added to the team and channel' is displayed. 
5. Search for the same guest user by last name, who is part of the team and channel and click Invite Guests and verify the error message 'This person is already a member of all the channels' is displayed. 
6. Search and add an existing guest by email, who is not part of the team and click Invite Guests and verify the success message 'This guest has been added to the team and channel.’ is displayed 
7. Search and add a new guest by email, who is not part of the team and click Invite Guests and verify the success message 'This guest has been added to the team and channel.’ is displayed

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-18042

### CODE REVIEW

File: e2e/cypress/support/api_commands.js [hunk 1]
```diff
@@ -714,3 +714,14 @@ Cypress.Commands.add('loginAsNewGuestUser', (user = {}, bypassTutorial = true) =
...
+/**
+ * Demote a Member to Guest directly via API
+ * @param {String} userId - The user ID
+ * All parameter required
+ */
+Cypress.Commands.add('demoteUser', (userId) => {
+    //Demote Regular Member to Guest User
+    const baseUrl = Cypress.config('baseUrl');
+    cy.externalRequest({user: users.sysadmin, method: 'post', baseUrl, path: `users/${userId}/demote`});
+});
```
Comment (saturninoabril) [inline e2e/cypress/support/api_commands.js]: 💯 . Since we have this now, it would be nice to change the other instance to use this as well. (Can be here or in separate PR)

**Comment by srkgupta** [comment]
Hi @prapti 

I have now added step to reload current page after each test to close any popup/modals left open. Please check now.

**Comment by prapti** [comment]
@srkgupta still seeing some error. This one reads a "Security Error".
